airsonic-main/src/main/java/org/airsonic/player/service/search/IndexManager.java

```diff
@@ -70,110 +72,165 @@
+     * Close Writer of all indexes and update SearcherManager.
+     * Called at the end of the Scan flow.
      */
-    @Deprecated
-    private File getIndexRootDirectory() {
-        return new File(SettingsService.getAirsonicHome(), LUCENE_DIR);
+    public void stopIndexing() {
+        Arrays.asList(IndexType.values()).forEach(i -> stopIndexing(i));
     }
 
     /**
-     * @deprecated It becomes unnecessary in the future
+     * Close Writer of specified index and refresh SearcherManager.
+     * @param type
      */
-    @Deprecated
-    private File getIndexDirectory(IndexType indexType) {
-        return new File(getIndexRootDirectory(), indexType.toString().toLowerCase());
-    }
+    private void stopIndexing(IndexType type) {
+
+        boolean isUpdate = false;
+        // close
+        try {
+            isUpdate = -1 == writers.get(type).commit();
+            writers.get(type).close();
+            writers.remove(type);
+            LOG.trace("Success to create or update search index : [" + type + "]");
+        } catch (IOException e) {
+            LOG.error("Failed to create search index.", e);
+        } finally {
+            FileUtil.closeQuietly(writers.get(type));
+        }
+
+        // refresh reader as index may have been written
+        if (isUpdate && searchers.containsKey(type)) {
+            try {
+                searchers.get(type).maybeRefresh();
+                LOG.trace("SearcherManager has been refreshed : [" + type + "]");
+            } catch (IOException e) {
+                LOG.error("Failed to refresh SearcherManager : [" + type + "]", e);
+                searchers.remove(type);
+            }
+        }
 
-    private IndexWriter createIndexWriter(IndexType indexType) throws IOException {
-        File dir = util.getDirectory.apply(LUCENE_DIR, indexType);
-        IndexWriterConfig config = new IndexWriterConfig(analyzerFactory.getAnalyzer());
-        return new IndexWriter(FSDirectory.open(dir.toPath()), config);
     }
 
-    public final void startIndexing() {
+    /**
+     * Return the IndexSearcher of the specified index.
+     * At initial startup, it may return null
+     * if the user performs any search before performing a scan.
+     * 
+     * @param indexType
+     * @return
+     */
+    public @Nullable IndexSearcher getSearcher(IndexType indexType) {
+        if (!searchers.containsKey(indexType)) {
+            synchronized (searchers) {
+                File indexDirectory = util.getDirectory.apply(getIndexRootDirectoryName(), indexType);
+                try {
+                    if (indexDirectory.exists()) {
+                        SearcherManager manager = new SearcherManager(FSDirectory.open(indexDirectory.toPath()), null);
+                        searchers.put(indexType, manager);
+                    } else {
+                        LOG.warn("{} does not exist. Please run a scan.", indexDirectory.getAbsolutePath());
```

Comment:
If you search before scan, warning.
Need to fix message level or log level?

Reply:
Warn should be fine here.